pkg/workloadmeta/collectors/internal/process/process.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package process
+
+import (
+	"context"
+
+	"github.com/benbjohnson/clock"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	dderrors "github.com/DataDog/datadog-agent/pkg/errors"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	workloadmetaExtractor "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+const collectorId = "local-process"
+
+// Used for testing
+var c *collector
+
+func init() {
+	// The process collector can run either when workloadmeta is in local or remote collector mode.
+	workloadmeta.RegisterCollector(collectorId, newProcessCollector)
+	workloadmeta.RegisterRemoteCollector(collectorId, newProcessCollector)
+}
+
+func newProcessCollector() workloadmeta.Collector {
+	// TODO: Inject config.Datadog via fx once collectors are migrated to components.
+	ddConfig := config.Datadog
+
+	wlmExtractor := workloadmetaExtractor.NewWorkloadMetaExtractor(ddConfig)
+
+	processData := checks.NewProcessData(ddConfig)
+	processData.Register(wlmExtractor)
+
+	c = &collector{
+		ddConfig:        ddConfig,
+		wlmExtractor:    wlmExtractor,
+		grpcServer:      workloadmetaExtractor.NewGRPCServer(ddConfig, wlmExtractor),
+		processData:     processData,
+		collectionClock: clock.New(),
+		pidToCid:        make(map[int]string),
+	}
+	return c
+}
+
+// Compile time check to ensure that `collector` implements `workloadmeta.Collector`.
+var _ workloadmeta.Collector = (*collector)(nil)
+
+type collector struct {
+	ddConfig config.ConfigReader
+
+	processData *checks.ProcessData
+
+	wlmExtractor *workloadmetaExtractor.WorkloadMetaExtractor
+	grpcServer   *workloadmetaExtractor.GRPCServer
+
+	pidToCid map[int]string
+
+	collectionClock clock.Clock
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
+	if enabled, err := Enabled(c.ddConfig); !enabled {
```

Comment:
Don't you mean to check `err` here?
```suggestion
	if enabled, err := Enabled(c.ddConfig); !err {
```

Reply:
Ah good point, I guess we don't need the enabled boolean